pgcli/pgexecute.py

```diff
@@ -263,15 +267,14 @@ def run(self, statement, pgspecial=None, on_error=ON_ERROR_RESUME):
                 _logger.error("traceback: %r", traceback.format_exc())
 
                 if (isinstance(e, psycopg2.OperationalError)
-                        or on_error == ON_ERROR_RAISE):
+                        or not exception_formatter):
                     # Always raise operational errors, regardless of on_error
                     # specification
                     raise
 
-                result = click.style(utf8tounicode(str(e)), fg='red')
-                yield None, None, None, result
+                yield exception_formatter(e)
```

Comment:
exception_formatter() should only format the exception, so this should be `yield None, None, None, exception_formatter(e)`.